packages/component-boilerplate/src/environment/UserIdentityContext.ts

```diff
@@ -150,7 +150,7 @@ export function useUserIdentity(): UserIdentity | undefined {
     | undefined;
 
   const validIdentity = userIdentityFromJwt(identity?.jwt);
-  if (!validIdentity) {
+  if (!validIdentity && identity?.jwt !== undefined) {
```

Comment:
This should instead be an update to `setUserIdentity` so that it only updates global context when the value changes.

Reply:
Please add `debug` statements